language-plutus-core/plutus-ir/Language/PlutusIR/Transform/Substitute.hs

```diff
@@ -17,15 +18,23 @@ import           Language.PlutusCore.Subst (substTyVar, typeSubstTyNames)
 
 import           Control.Lens
 
+import           Data.Maybe
+
 -- Needs to be different from the PLC version since we have different Terms
 -- | Replace a variable using the given function.
-substVar :: (name -> Maybe (Term tyname name uni a)) -> Term tyname name uni a -> Term tyname name uni a
-substVar nameF (Var _ (nameF -> Just t)) = t
-substVar _ t                             = t
+substVar :: (name -> Maybe (Term tyname name uni a)) -> Term tyname name uni a -> Maybe (Term tyname name uni a)
+substVar nameF (Var _ (nameF -> Just t)) = Just t
```

Comment:
Would it make sense to write the first clause as

```haskell
substVar nameF (Var _ name) = nameF name
```

?

Reply:
Yep